beetle/iroh-bitswap/src/lib.rs

```diff
@@ -423,23 +432,17 @@ impl<S: Store> NetworkBehaviour for Bitswap<S> {
             }
             libp2p::swarm::FromSwarm::DialFailure(event) => {
                 if let Some(peer_id) = event.peer_id {
-                    if let DialError::Denied { cause: _ } = event.error {
-                        // TODO check that the denied cause is because of a connection limit.
-                        //if let Ok(connection_limits::Exceeded { .. }) = cause.downcast() {
-                        self.pause_dialing = true;
-                        self.set_peer_state(&peer_id, PeerState::Disconnected);
-                        //}
-                    } else {
-                        self.set_peer_state(&peer_id, PeerState::DialFailure(Instant::now()));
-                    }
-
-                    trace!("inject_dial_failure {}, {:?}", peer_id, event.error);
-                    let dials = &mut self.dials.lock().unwrap();
-                    if let Some(mut dials) = dials.remove(&peer_id) {
-                        while let Some((_id, sender)) = dials.pop() {
-                            let _ = sender.send(Err(event.error.to_string()));
+                    match event.error {
+                        DialError::Denied { cause: _ } => {
+                            // TODO check that the denied cause is because of a connection limit.
+                            //if let Ok(connection_limits::Exceeded { .. }) = cause.downcast() {
+                            self.pause_dialing = true;
+                            //}
                         }
-                    }
+                        _ => {}
+                    };
+                    trace!(%peer_id, error = %event.error,  "report dial failure");
```

Comment:
Will this give us count of the different `event.error`s happening here?

Reply:
No, this will just capture a span of the event error. We can add an explicit metric if needed.